src/tests/streams/streamSigning.test.ts

```diff
@@ -27,8 +29,79 @@ describe ('stream signing', function () {
 		});
 	});
 
-	describe ('signature verification', () => {
+	describe('transfer the signature', () => {
+		let publicKey: VirgilPublicKey;
+		let privateKey: VirgilPrivateKey;
+		let signature: Buffer;
+		let inputChunks: Buffer[];
+
+		beforeEach(async () => {
+			({ privateKey, publicKey } = createVirgilKeyPair());
+			const streamSigner = new VirgilStreamSigner();
+			const input = Buffer.alloc(3 * 1000).fill('foo');
+			inputChunks = splitIntoChunks(input, CHUNK_SIZE);
+
+			for await (const chunk of createAsyncIterable(inputChunks)) {
+				streamSigner.update(chunk);
+			}
+
+			signature = streamSigner.sign(privateKey);
+		});
+
+		const transferSignature = async (encoding?: StringEncoding) => {
+			assert.isTrue(Buffer.isBuffer(signature));
+
+			const streamCipher = new VirgilStreamCipher(
+				publicKey,
+				encoding ? signature.toString(encoding) : signature,
+				encoding
+			);
+			const encryptedBuffer: Buffer[] = [];
+
+			encryptedBuffer.push(streamCipher.start());
+
+			for await (const chunk of createAsyncIterable(inputChunks)) {
+				encryptedBuffer.push(streamCipher.update(chunk));
+			}
+			encryptedBuffer.push(streamCipher.final());
+
+			const streamDecipher = new VirgilStreamDecipher(privateKey);
+			const decryptedBuffer: Buffer[] = [];
+
+			for await (const chunk of createAsyncIterable(encryptedBuffer)) {
+				decryptedBuffer.push(streamDecipher.update(chunk));
+			}
+
+			decryptedBuffer.push(streamDecipher.final(false));
+			let transferredSignature: Buffer | string = streamDecipher.getSignature()!;
+			if (encoding) transferredSignature = transferredSignature.toString(encoding);
+			streamDecipher.dispose();
+			assert.exists(transferredSignature);
+			const streamVerifier = new VirgilStreamVerifier(transferredSignature!, encoding);
+
+			for await (const chunk of createAsyncIterable(inputChunks)) {
+				streamVerifier.update(chunk);
+			}
+
+			encoding
+				? assert.isString(transferredSignature)
+				: assert.isTrue(Buffer.isBuffer(transferredSignature));
```

Comment:
This logic is not needed, you should not put business logic in the test. `getSignature` always returns signature as `Buffer` and you *only* need to verify that.